docs/DeveloperGuide.md

```diff
@@ -2,33 +2,249 @@
+handled by parsing these inputs into Commands using a Parser . Below is a UML Diagram
+showing how the `Parser` classes and the `Command` classes interact. Unless the `Command` is
+an `ExitCommand` or `HelpCommand` , the `Command` will be executed to perform an action to
+either the `ScheduleManager` or `ModuleManager` or both.
+
+![](Images/3.3Command.JPG)
+
+### Managers
+The application consists of two managers, the `ScheduleManager` and `ModuleManager`. The
+`ScheduleManager` will handle the storing of `Task` in each day. The `ModuleManager` will
+handle storing the task of each `Module`. Below is a UML Diagram showing some of the key
+methods and properties of the `ScheduleManager` and `ModuleManager` classes
+
+![](Images/3.4Managers.JPG)
+
+Some Design Considerations on how to store the `Task` in the `ScheduleManager`:
+- Use of a `TreeMap` to store the date as keys and `ArrayList` as values.
+    - Pros: Use of a `TreeMap` allows iteration of dates as the key to be iterated in order, which
+                might be useful in further development of implementation of the use of the `ScheduleManager`.
+                Searching for a date can also be done in O(logn) similarly to a `HashMap`.
+    - Cons: Storing the dates that have past and also storing of many dates might be heavy in
+                terms of space.
+- Use of a `HashMap` to store the date as keys and `ArrayList` as values
+    - Pros: Allows tracking of tasks with respect to the dates as keys.
+    - Cons: Does Not allow iteration in order as when desired as compared to a `TreeMap`.
+- Creation of a new class which contains a schedule for each day, and using a `TreeSet`.
+      to contain this new class.
+    - Pros: Making use of more oop design.
+    - Cons: Might be tricky to implement and may need more resources.
+    
+### Storage
+The application will save all of the user inputs by using a `Storage` class, which is designed as
+singleton.
+
+![](Images/3.5Stroage.JPG)
+
+It exports the data in the application and converts it into the form of a txt file. This allows users
+to retain their saved data after exiting the app. More elaboration of how this will be done will be shown later, 
+where a sequence diagram will show how this process works.
+
+## 4. Implementation
+
+This section describes some noteworthy details on how certain features are
+implemented.
+
+### Add Feature
+
+The add implementation allows the user to add a `Lesson`, `Event` or `Deadline`. This is facilitated
+by implementing the ScheduleManager and ModuleManager, which stores all of the `Task` in
+them. The `ScheduleManager` helps to save these tasks into the dates according to the date of
+the `Task` as stated by the User. The `ModuleManager` will also add the `Task` into the respective module
+that it belongs to. Below is a sequence diagram which shows how adding of a
+`Lesson` works. Adding of a Event or a Deadline is also in a similar fashion.
+
+![](Images/4.1Add.JPG)
```

Comment:
Will any of these objects be terminated? Can add cross into sequence diagram to illustrate that.